Integrations/Forescout/Forescout.py

```diff
@@ -0,0 +1,678 @@
+        title = 'Index of Host Properties'
+        human_readable = tableToMarkdown(title, content, removeNull=True)
+        return_outputs(readable_output=human_readable, outputs=context, raw_response=data)
+
+
+def get_policies():
+    url_suffix = '/api/policies'
+    headers = create_web_api_headers()
+    response = http_request('GET', url_suffix, headers=headers, resp_type='response')
+    return response
+
+
+def get_policies_command():
+    response = get_policies()
+    data = response.json()
+    content = format_policies_data(data)
+    readable_content = deepcopy(content)
+    for policy in readable_content:
+        readable_rules = []
+        for rule in policy.get('Rule', []):
+            readable_rules.append(dict_to_formatted_string(rule))
+        policy['Rule'] = readable_rules
+    context = {'Forescout.Policy(val.ID && val.ID === obj.ID)': content}
+    title = 'Forescout Policies'
+    human_readable = tableToMarkdown(title, readable_content, removeNull=True)
+    return_outputs(readable_output=human_readable, outputs=context, raw_response=data)
+
+
+def update_lists(args={}):
+    update_type = args.get('update_type', '')
+    lists = args.get('lists', '')
+    req_body = create_update_lists_request_body(update_type, lists)
+    data = ET_PHONE_HOME.tostring(req_body, encoding='UTF-8', method='xml')
+    url_suffix = '/fsapi/niCore/Lists'
+    response = http_request('POST', url_suffix, headers=DEX_HEADERS, auth=DEX_AUTH, data=data, resp_type='response')
+    return response
+
+
+def update_lists_command():
+    args = demisto.args()
+    response = update_lists(args)
+    resp_xml = ET_PHONE_HOME.fromstring(response.content)
+    code = [child.text for child in resp_xml.iter() if child.tag == 'CODE'][0]
+    msg = [child.text for child in resp_xml.iter() if child.tag == 'MESSAGE'][0]
+    result_msg = f'{code}: {msg}'
+    demisto.results(result_msg)
+
+
+def update_host_properties(args={}):
+    host_ip = args.get('host_ip', '')
+    update_type = args.get('update_type', '')
+    properties = args.get('properties', '')
+    composite_property = args.get('composite_property', '').replace('\'', '"')
+    req_body = create_update_hostproperties_request_body(host_ip, update_type, properties, composite_property)
+    data = ET_PHONE_HOME.tostring(req_body, encoding='UTF-8', method='xml')
+    url_suffix = '/fsapi/niCore/Hosts'
+    response = http_request('POST', url_suffix, headers=DEX_HEADERS, auth=DEX_AUTH, data=data, resp_type='response')
+    return response
+
+
+def update_host_properties_command():
+    args = demisto.args()
+    update_type = args.get('update_type', '')
+    properties = args.get('properties', '')
+    response = update_host_properties(args)
+
+    # Because the API has an error and says it deletes multiple things when it only deletes one
+    # have to take care of it behind the curtains
+    if update_type == 'delete':
+        args_copy = deepcopy(args)
```

Comment:
why do we need a copy?

Reply:
otherwise when I manipulate members of the dict, e.g. when I alter the appearance to look nice in the war room readable output, it will also be changed in the context outputs.